src/main/java/org/ibp/api/rest/role/RoleValidator.java

```diff
@@ -0,0 +1,80 @@
+package org.ibp.api.rest.role;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.Transformer;
+import org.generationcp.middleware.manager.api.WorkbenchDataManager;
+import org.generationcp.middleware.pojos.workbench.Permission;
+import org.generationcp.middleware.pojos.workbench.RoleTypePermission;
+import org.ibp.api.exception.ApiRequestValidationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.MapBindingResult;
+
+import java.util.Collection;
+import java.util.HashMap;
+
+@Component
+public class RoleValidator {
+
+	@Autowired
+	private WorkbenchDataManager workbenchDataManager;
+
+	private BindingResult errors;
+
+	public void validateRoleGeneratorInput(final RoleGeneratorInput roleGeneratorInput) {
+
+		final BindingResult errors = new MapBindingResult(new HashMap<String, String>(), RoleGeneratorInput.class.getName());
+
+		if (roleGeneratorInput.getName().length() > 100) {
+			errors.reject("role.name.lenght");
+		}
+
+		if (this.workbenchDataManager.getRoleByName(roleGeneratorInput.getName()) != null) {
+			errors.reject("role.name.already.exists");
+		}
+
+		if (this.workbenchDataManager.getRoleByDescription(roleGeneratorInput.getDescription()) != null) {
+			errors.reject("role.description.already.exists");
+		}
+
+		if (!roleGeneratorInput.getDescription().isEmpty()) {
+			if (roleGeneratorInput.getDescription().length() > 255) {
+				errors.reject("role.description.lenght");
+			}
+		}
+
+		if (roleGeneratorInput.getRoleType() == null) {
+			errors.reject("role.roletype.null");
+		}
+
+		if (this.workbenchDataManager.getRoleType(roleGeneratorInput.getRoleType()) == null) {
+			errors.reject("role.roletype.does.not.exist");
+		}
+		else {
+			for (final Integer permissionId : roleGeneratorInput.getPermissions()) {
+				final Permission permission = this.workbenchDataManager.getPermission(permissionId);
+				if (permission == null) {
+					errors.reject("role.permission.does.not.exist");
+				}
+
+				final Collection result = CollectionUtils.collect(permission.getRoleTypePermissions(), new Transformer() {
```

Comment:
@vmaletta does it need to be done in a for? Can be done once?

Reply:
How?